### PR TITLE
Add IP Search for Mods, Deep IP Search for Admins

### PR DESCRIPTION
**Pull request by bls1999**
You're welcome @EternalUF :joy:

In all seriousness, this will make our jobs a LOT easier.

### CODE REVIEW

**Comment by SteelChivalry** [comment]
@jacob-grahn You should keep to your word and stay as much on other side as possible about this. This could be very bad if mod account gets compromised again. They already have many searching tools, but this may do more harm than good.

**Comment by bls1999** [comment]
@SteelChivalry ah, okay, so now not only did you submit a malicious pull request, but you're attempting to advocate against a tool that will allow us so find rule-breakers in a much easier fashion by account/IP association.  Seems legit!!!  Give me a break :joy: :joy: :joy:

**Comment by bls1999** [comment]
Additionally, all of this information is already available if you know how to look for it.  It just takes a lot longer.  This will shorten the process substantially and aid our moderation.

**Comment by SteelChivalry** [comment]
No. I did not submit malicious pull request.

@Jmarroqu and others will agree that staff team is trying to stalk users of game Platform Racing 2 now. Not cool.

**Comment by bls1999** [comment]
Didn't make a malicious pull request, you say?  Hmm… let's take a trip down memory lane…

https://github.com/jacob-grahn/platform-racing-2-server/pull/330

**Comment by bls1999** [comment]
"@steelchivalry closed this pull request." Hmm, I wonder why… :thinking:

**Comment by SteelChivalry** [comment]
We already have seen abuse of features from mods like @aaaaaa123456789 so why do we need to see same thing happen again?

**Comment by aaaaaa123456789** [comment]
I'll let you point out any instance when I abused IP searching tools. Have fun.

That being said, I always wondered why these tools weren't available to begin with. They were on JV (and for all mods, not just admins), which was very useful at times.

**Comment by bls1999** [comment]
@aaaaaa123456789 there's an IP info utility in there for both mods and admins.  The mod one searches by IP/register IP currently associated with an account while the admin one searches recent logins as well.

Additionally, you never abused any mod tools, so that's definitely a red herring posted by Stardox up there.

**Comment by bls1999** [comment]
@jacob-grahn I'll also add that this is much less dangerous than adding a search by email tool.

**Comment by SteelChivalry** [comment]
I don't want to see feature like this getting abused by ambitious staff. Mods now are too worried about accounts and not being helpful to players of Platform Racing 2. Also there was situation when Bls1999 logged in UnknownAX and took account from him unfairly. Admins are abusing by stealing handles from others.

**Comment by bls1999** [comment]
@SteelChivalry I love when people talk about things they don't know anything about!  Not only is that untrue, but it's completely unrelated to anything on this pull request.  You created multiple accounts solely for the purpose of sneaking an exploit into the server and sabotaging any attempts for us to get more intuitive mod tools.  Delete your account.

**Comment by EternalUF** [comment]
Much like email searching before the tool was merged, IP searching is already possible. Just extremely tidious. The great thing about this PR is the distinction from the Mod IP search tool and the Admin search tool. Which would allow searching based off their respected player search level which is nice. With the powers Admins and Moderators have now, I see no issue with abuse, and don’t plan on it with the addition of this IP search tool. Like Ax6 said, I’m surprised this wasn’t added sooner.

**Comment by aaaaaa123456789** [comment]
Someone once very intelligently said that when you give multiple and unrelated reasons for something you want, throwing in a new one every time the previous one is rejected, you don't actually believe your arguments — you're just testing arguments to see if something works.

**Comment by SteelChivalry** [comment]
This is why staff like Bls1999 and other moderator should not have this feature. Im not testing arguments, it's valid points.

**Comment by aaaaaa123456789** [comment]
You're not even _giving_ arguments at this point; you're just crying "abuse!!!" when there are zero known instances of it...

**Comment by fourst4r** [comment]
> so that's definitely a red herring posted by Stardox up there.

@bls1999 You don't need to drag an innocent man's name through the dirt. Stardox is not the one submitting the malicious pull requests, any idiot could see that.

**Comment by bls1999** [comment]
@fourst4r we've determined that it is in fact Stardox.

**Comment by aaaaaa123456789** [comment]
While I see no evidence backing his claim, I see none disproving it,
either. Why are you so sure?

**Comment by fourst4r** [comment]
@bls1999 Discord screenshots are not valid proof.

**Comment by bls1999** [comment]
@fourst4r the proof we have is not a discord screenshot.

**Comment by fourst4r** [comment]
@bls1999 Prove me wrong.

**Comment by bls1999** [comment]
@fourst4r you're not privy to confidential information used in the determination of bans.  Either way, whether or not you believe it's Stardox has nothing to do with this pull.

**Comment by fourst4r** [comment]
@bls1999 Yeah that's what I thought. Soiling the name of a well respected player and refusing to give any evidence also has nothing to do with this pull. 
But we should believe you because you have such a good track record of _banning the right people, right?_ All those Adak alts that weren't Adak, those botters that weren't banned, those regular players that were banned for botting... you can't blame people for being skeptical. 
You should be encouraging people to disprove evidence, not vice versa, especially considering Stardox has never shown any inkling of maliciousness to anyone.

@aaaaaa123456789 For one, Stardox is not stupid, he is not the type of person to send malicious pull requests on GitHub (for reasons other than obviously nobody would blindly merge pull requests from unknown people). [He is well-versed in PR2's inner workings,](https://github.com/Stardox/PR2PS) so why would he need to _create_ exploits, I'm sure he has discovered his fair share in this decade old game... so what in the world could his motive be?
Bls is trying to force jigsaw pieces that don't go together.

**Comment by LP-PR2** [comment]
@fourst4r he needs as much information of people as possible so he can dox them when they piss him off, just like he doxed Adak with Fame.

**Comment by ShakeJunt01** [comment]
( ͡° ͜ʖ ͡°)

**Comment by FirefightPR2** [comment]
This game's staff team is full of clowns now. They don't know what is and isn't valid proof. I recently got banned for "stealing accounts", but I never stole anyone's accounts. I have had a good track record for a long time, and I wouldn't just come back to PR2 after being inactive and start stealing people's accounts. I asked for proof, but I didn't receive anything whatsoever. Is our admin Bls too afraid to show me his "proof" because he isn't confident that it's legit? 

Bls accepts Discord screenshots as proof. I and many others have been banned because people have presented Discord screenshots with "proof" of us breaking rules on PR2. It's obvious that people are framing me and others. Frankly, I'm not very happy that someone spilled the beans that I was botting, but it's clear to me that all of this framing for stuff I didn't do is fueled by obsession and jealousy. They're just trying to give innocent people like us even more hell. A toddler could make a ****ing fake conversation or use inspect element on Discord. It's horrendous that we have to serve year bans for things we didn't do.

Now, intelligent admin Bls, do you really think Stardox is that stupid? He's programmed a private server before, which takes a lot of effort. This was before the source code to this server was released. Look at "Stardox's" attempts to resurrect exploits. I mean, really? Something must be wrong with you if you think he became 5x dumber. A person doesn't just lose half of his IQ. Get this BS straight.

**Comment by aaaaaa123456789** [comment]
>This game's staff team is full of clowns now. They don't know what is and isn't valid proof. I recently got banned for "stealing accounts", but I never stole anyone's accounts. I have had a good track record for a long time, and I wouldn't just come back to PR2 after being inactive and start stealing people's accounts. I asked for proof, but I didn't receive anything whatsoever. Is our admin Bls too afraid to show me his "proof" because he isn't confident that it's legit?

Considering I have solid proof that you're not the real Firefight at all despite not having access to your IP history, this is just hilarious.

**Comment by bls1999** [comment]
@aaaaaa123456789 hmm... "Joined 20 minutes ago" :joy: give me a break

**Comment by FirefightPR2** [comment]
@aaaaaa123456789 What proof? Oh, Discord screenshots? Give it a break.

**Comment by bls1999** [comment]
@FirefightPR2 prove me wrong and appeal the ban.

**Comment by aaaaaa123456789** [comment]
@FirefightPR2 I wouldn't call screenshots "solid proof". I've known how to modify a screenshot for literally decades.

**Comment by fourst4r** [comment]
> I wouldn't call screenshots "solid proof". I've known how to modify a screenshot for literally decades.

@bls1999 do you need me to link you an image manipulation tutorial? There's tons on YouTube.

File: http_server/queries/staff/admin/users_select_by_ip_expanded.php [hunk 1]
```diff
@@ -12,6 +12,8 @@ function users_select_by_ip_expanded($pdo, $search_ip)
           LEFT JOIN recent_logins rl ON u.user_id = rl.user_id
         WHERE
           :search_ip IN (u.ip, u.register_ip, rl.ip)
+        ORDER BY
+          time DESC
```
Comment (aaaaaa123456789) [inline http_server/queries/staff/admin/users_select_by_ip_expanded.php]: `u.time`, not `time`

Comment (bls1999) [reply http_server/queries/staff/admin/users_select_by_ip_expanded.php]: Nice catch, thanks.

**Comment by aaaaaa123456789** [comment]
You ninja, you fixed what I was about to point out.

**Comment by bls1999** [comment]
Ha! Every dog has its day. :man_shrugging:

**Pull request merged**